libexec/python/cli.py

```diff
@@ -112,11 +115,12 @@ def main():
            sys.exit(1)
        logger.info("Root file system defined by env variable as %s", singularity_rootfs)
 
-    # Does the registry require a token?
-    doauth = True
-    if args.notoken == True:
-       doauth = False
-    logger.info("Do registry authentication set to %s", doauth)
+    # Does the registry require authentication?
+    auth = None
+    if args.username is not None and args.password is not None:
```

Comment:
Can we move this guy into a function in docker/utils.py that takes a username,password, and returns an updated auth object in the same fashion? This will be useful in case we need to sneak it in other places at some point - we'd want them all to change based on pinging the same function.